src/mappings/content/utils.ts

```diff
@@ -651,3 +658,244 @@ export function encodeAssets(assets: StorageAssetsRecord | undefined): Uint8Arra
+  const winningLink = await openAuctionWonLink(em, videoId)
+  const losingLink = await openAuctionLostLink(em, videoId)
+  return {
+    won: (memberHandle: string) =>
+      new OpenAuctionWon({
+        recipient: new MemberRecipient({
+          memberHandle,
+        }),
+        data: new NotificationData({
+          linkPage: winningLink,
+          text: openAuctionBidWonText(videoTitle),
+        }),
+      }),
+    lost: (memberHandle: string) =>
+      new OpenAuctionLost({
+        recipient: new MemberRecipient({
+          memberHandle,
+        }),
+        data: new NotificationData({
+          linkPage: losingLink,
+          text: openAuctionBidLostText(videoTitle),
+        }),
+      }),
+  }
+}
+
+export const englishAuctionNotifiers = async (
+  { em, videoId }: PageLinkData,
+  videoTitle: string
+): Promise<AuctionNotifiers> => {
+  const winningLink = await timedAuctionWonLink(em, videoId)
+  const losingLink = await timedAuctionLostLink(em, videoId)
+  return {
+    won: (memberHandle: string) =>
+      new EnglishAuctionWon({
+        recipient: new MemberRecipient({
+          memberHandle,
+        }),
+        data: new NotificationData({
+          linkPage: winningLink,
+          text: timedAuctionBidWonText(videoTitle),
+        }),
+      }),
+    lost: (memberHandle: string) =>
+      new EnglishAuctionLost({
+        recipient: new MemberRecipient({
+          memberHandle,
+        }),
+        data: new NotificationData({
+          linkPage: losingLink,
+          text: timedAuctionBidLostText(videoTitle),
+        }),
+      }),
+  }
+}
+
+export async function notifyChannelOwner(
+  overlay: EntityManagerOverlay,
+  nftOwner: NftOwner,
+  notificationType: (channelTitle: string) => NotificationType,
+  event: Event
+): Promise<void> {
+  if (nftOwner.isTypeOf === 'NftOwnerChannel') {
+    const channelId = nftOwner.channel
+    const channel = await overlay.getRepository(Channel).getByIdOrFail(channelId)
+    const nftOwnerAccount = await getAccountForMember(overlay.getEm(), channel.ownerMemberId)
+    await addNotification(
+      overlay.getEm(),
+      nftOwnerAccount,
+      notificationType(channel.title || ''),
+      event
+    )
+  }
+}
+
+export async function addRoyaltyPaymentNotification(
+  overlay: EntityManagerOverlay,
+  channelId: string | undefined | null,
+  notificationType: (channelTitle: string) => NotificationType,
+  event: Event
+): Promise<void> {
+  if (channelId) {
+    const channel = await overlay.getRepository(Channel).getByIdOrFail(channelId)
+    const creatorAccount = await getChannelOwnerAccount(overlay.getEm(), channel)
+    await addNotification(
+      overlay.getEm(),
+      creatorAccount,
+      notificationType(channel.title || ''),
+      event
+    )
+  }
+}
+
+export function computeRoyalty(royaltyPct: number, price: bigint): bigint {
+  const royaltyDecimal = royaltyPct / 100
+  const royaltyPrice = Math.floor(royaltyDecimal * Number(price))
```

Comment:
I wonder if there could be any data loss because of the wrong conversion from bigint to number (and if not, then why would you use bigint, if all of the numbers could be included in number?)